src/aws-cpp-sdk-core/include/aws/core/auth/CustomSSOCredentialsProvider.h

```diff
@@ -0,0 +1,34 @@
+/**
+* Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+#pragma once
+
+#include <aws/core/auth/SSOCredentialsProvider.h>
+#include <aws/core/client/ClientConfiguration.h>
+
+namespace Aws {
+    namespace Auth {
+        class AWS_CORE_API CustomSSOCredentialsProvider : public SSOCredentialsProvider {
```

Comment:
this class isnt needed we should just be adding a constructor the `SSOCredentialsProvider` itself, no need to create a indirection.